### PR TITLE
Separate developer and user agent skills

### PR DESCRIPTION
**Pull request by drbenvincent**
## Summary

- Move user-facing CausalPy agent skills from `.github/skills/` into `causalpy/skills/` so they live with the source tree and can be distributed as user-facing guidance via Decision AI Hub.
- Keep developer/maintainer skills in `.github/skills/`, with `.cursor/skills`, `.claude/skills`, and `.agents/skills` continuing to symlink there for in-repo agent workflows.
- Add a `causalpy/skills/README.md` and update `AGENTS.md`, `README.md`, and Cursor command pointers to document the split between developer skills and user-facing skills.
- Remove the previous installer/CLI approach; end-user distribution is handled outside the package via Decision AI Hub.

## Changed Skill Layout

- `.github/skills/` now contains developer workflow skills only: environment setup, PR workflows, issue handling, review, and marimo workflows.
- `causalpy/skills/` now contains user-facing CausalPy usage skills: `designing-experiments`, `performing-causal-analysis`, and `running-placebo-analysis`.
- `performing-causal-analysis` includes method references for difference-in-differences, interrupted time series, synthetic control, and regression discontinuity.
- The previous `loading-datasets` user skill was removed because `causalpy.load_data()` already documents available datasets directly.

## Review Notes

- The user-facing skills are intentionally not auto-discovered in the contributor/developer context; this keeps CausalPy maintainer agents focused on repo workflow skills rather than end-user analysis guidance.
- If an agent is explicitly pointed at `causalpy/skills/`, the folder has a standard skill structure: each skill has a `SKILL.md` with YAML frontmatter and local reference links resolve.
- This PR does not try to perfect the skill content itself; deeper review of skill quality can happen in a follow-up PR.

## Test Plan

- [x] Confirm tracked skill files are split correctly between `.github/skills/` and `causalpy/skills/`.
- [x] Validate `causalpy/skills/*/SKILL.md` files have frontmatter with `name` and `description`.
- [x] Validate local reference links from user-facing `SKILL.md` files resolve.
- [x] `prek` passes in CI.
- [x] Python tests pass in CI for 3.11 and 3.14.
- [x] Notebook checks pass in CI.
- [x] Read the Docs preview build passes.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: causalpy/skills/_platforms/_base.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        """Return the version string from an existing install, or ``None``."""
+
+    @staticmethod
+    def _extract_version(text: str) -> str | None:
+        """Extract the CausalPy version from a version-stamp comment in *text*."""
+        m = re.search(r"<!-- causalpy-skills v([\d.]+\S*) -->", text)
+        return m.group(1) if m else None
+
+    def _stamp(self, content: str) -> str:
+        """Prepend version stamp to content."""
```
[the remark this answers is not in the document]
Comment (drbenvincent) [reply causalpy/skills/_platforms/_base.py]: Good catch. Fixed in 5934f90c — `_stamp()` now uses a regex to detect YAML frontmatter (`---\n...\n---\n`) and inserts the version stamp comment after the closing delimiter. Content without frontmatter still gets the stamp prepended as before. Added a test (`test_version_stamp_after_frontmatter`) that verifies the output starts with `---` and the stamp appears after the frontmatter block.

File: causalpy/skills/_platforms/_base.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        return f"{self.version_stamp}\n{content}"
+
+    def _rmtree_if_exists(self, path: Path) -> bool:
+        """Remove *path* (file or directory) if it exists. Return whether it existed."""
+        if path.exists():
+            if path.is_dir():
+                shutil.rmtree(path)
+            else:
+                path.unlink()
+            return True
```
[the remark this answers is not in the document]
Comment (drbenvincent) [reply causalpy/skills/_platforms/_base.py]: Good catch — fixed in 5934f90c. `_rmtree_if_exists()` now:

1. Checks `path.is_symlink()` first and unlinks without following
2. For real paths, validates `resolved.is_relative_to(self.project_dir.resolve())` and raises `ValueError` if the path would escape the project directory
3. Only calls `shutil.rmtree()` on verified real directories within the project

Added two tests: `test_symlink_unlinked_without_following` (verifies symlink target is preserved) and `test_refuses_path_outside_project` (verifies `ValueError` is raised).

**Review by drbenvincent — COMMENTED** [review]
For the record: I’m gpt-5.5 acting on behalf of @drbenvincent.

## Findings

- **Must fix** `causalpy/skills/_platforms/_base.py`: installed skill files get a version comment before YAML frontmatter. My targeted install check produced `<!-- causalpy-skills v0.8.0 -->` as line 1 and `---` as line 2. Cursor/Claude/Windsurf-style skills commonly require frontmatter at byte 0, so this risks making the shipped user skills invisible to the agents they target. Put the stamp inside frontmatter, after the closing frontmatter block, or in a separate managed metadata file.

- **Must fix** `causalpy/skills/_installer.py`: platform auto-detection treats any `.github/` directory as Copilot. That means `causalpy skills install` in most GitHub repos can silently create or rewrite `.github/copilot-instructions.md`, even when the user is only using Cursor/Claude/etc. For developer repos this is especially risky because `.github` usually means CI, not Copilot. Detect only an existing Copilot-specific file, or require `--platform copilot`.

- **Must fix** `docs/source/conf.py`, `Makefile`, `.readthedocs.yaml`: `html_extra_path` includes `llms.txt`, but the file is gitignored and RTD invokes Sphinx directly, not `make llms-txt`. The advertised stable hosted `llms.txt` URL therefore depends on an unrun generation step. Generate it during the docs build, commit it intentionally, or wire RTD/build config to create it before Sphinx reads `html_extra_path`.

- **Must fix** `causalpy/skills/_installer.py` and platform adapters: the new public CLI writes files, uninstalls files, mutates Copilot instructions, checks versions, and auto-detects platforms, but there are no tests under `causalpy/tests/` for this surface. Add `tmp_path` pytest coverage for install/uninstall/check/list, frontmatter preservation, Copilot detection, and multi-platform behavior.

- **Should fix** `causalpy/skills/_installer.py`: `check_version()` returns the first installed version it finds. In a multi-platform repo, it can report “up to date” while another installed platform is stale. Since version-locking is the main value proposition for agent users, the check should report per-platform status.

- **Should fix** `causalpy/skills/loading-datasets/SKILL.md` and `causalpy/skills/performing-causal-analysis/SKILL.md`: moving these into the wheel makes them user-facing product docs for agents. The dataset skill omits many public datasets, and the analysis skill advertises RD but only links DiD/ITS/SC references. Before release, audit the skill content against the current API so agents don’t teach stale or partial workflows.

## Merge Readiness

Verdict: request changes while draft.

Branch status: GitHub reports mergeable, draft, and blocked. CI was still pending when checked; `prek` and pre-commit were passing, several tests/docs jobs were pending.

## PR Summary

The design direction is good: separating maintainer/developer skills from packaged user skills should reduce agent-context pollution for CausalPy contributors while giving end users version-matched agent guidance. The implementation needs tightening around the actual user-agent install path, especially frontmatter preservation, non-surprising platform detection, docs generation, and tests.

## Test Evidence

I ran targeted checks in the `CausalPy` conda env for `list_skills()`, Cursor install/check/uninstall, and `.github`-only auto-detection. Those confirmed the frontmatter stamp problem and the broad Copilot detection behavior. Full local tests were not run; remote CI was still pending.

File: Makefile [hunk 3]
```diff
@@ -43,6 +43,9 @@ html: ## Build HTML documentation with Sphinx
 run_notebooks_full: ## Re-execute all notebooks and save outputs in place (slow)
 	python scripts/run_notebooks/runner.py --full
 
+llms-txt: ## Generate llms.txt from user skills for the docs site
+	python -m causalpy.skills._generate_llms_txt docs/source/llms.txt
+
```
[the remark this answers is not in the document]
Comment (drbenvincent) [reply Makefile]: Good point — fixed in 5934f90c. The `html` target now declares `llms-txt` as a prerequisite, so `make html` automatically generates `docs/source/llms.txt` before running Sphinx.

File: causalpy/skills/_installer.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    for name in _SKILL_DIRS:
+        skills_data[name] = _read_skill_tree(name)
+
+    result: dict[str, list[Path]] = {}
+    for plat in platforms:
+        adapter = _PLATFORM_ADAPTERS[plat](project, VERSION_STAMP)
+        result[plat] = adapter.install(skills_data)
+
+    return result
+
```
[the remark this answers is not in the document]
Comment (drbenvincent) [reply causalpy/skills/_installer.py]: Added `causalpy/tests/test_skills_installer.py` in 5934f90c with 27 tests covering:

- `list_skills()` catalogue
- `detect_platforms()` auto-detection for all platforms (Cursor, Claude, Copilot, Windsurf) plus edge cases
- `install()` for Cursor (directory structure, frontmatter preservation, reference files), generic (llms.txt output), and Copilot (create, append, idempotent replace)
- `uninstall()` for all three adapter styles plus no-op case
- `check_version()` round-trip
- Symlink safety (unlink without following, refuse paths outside project)
- Error handling (nonexistent project dir)

All tests use `tmp_path` and do not require the CausalPy conda env.

**Comment by drbenvincent** [comment]
## Addressing GPT 5.5 review (219ed954)

| Finding | Status | Commit |
|---------|--------|--------|
| **Must fix** — version stamp before frontmatter | Fixed previously | 5934f90c |
| **Must fix** — `.github/` treated as Copilot | Fixed previously (narrowed to require `copilot-instructions.md`) | 2ee21c22 |
| **Must fix** — RTD won't generate `llms.txt` | Fixed — `conf.py` now imports and runs `generate()` when the file is missing, so Sphinx builds (including RTD) produce it without `make llms-txt` | 219ed954 |
| **Must fix** — no tests | Fixed previously — 28 tests in `test_skills_installer.py` | 5934f90c |
| **Should fix** — `check_version()` returns first match | Fixed — now returns `dict[str, str \| None]` mapping each platform to its installed version; CLI shows per-platform status | 219ed954 |
| **Should fix** — skill content gaps | Fixed — `loading-datasets` now lists all 21 datasets (8 synthetic + 13 CSV); `performing-causal-analysis` now includes RD reference with full API docs | 219ed954 |
